src/SecureStore.CyberArkCCP/CyberArkCCPSecureStore.cs

```diff
@@ -0,0 +1,291 @@
+            Exception normalizationException = null;
+            try
+            {
+                string obj;
+                key = key.Replace('\\', '-');
+                if (string.IsNullOrEmpty(ctx.Folder))
+                {
+                    obj = $"{key}";
+                }
+                else
+                {
+                    obj = $"{ctx.Folder}\\{key}";
+                }
+
+                var uri = new Uri(ctx.URL + $"/AIMWebService/api/Accounts?appid={ctx.ApplicationId}&safe={ctx.Safe}&object={obj}");
+                var response = await _httpClient.GetAsync(uri);
+                var content = await response.Content.ReadAsStringAsync();
+                var result = JsonConvert.DeserializeObject<CyberArkCCPPassword>(content);
+
+                if (!string.IsNullOrEmpty(result.PasswordChangeInProcess))
+                {
+                    if (result.PasswordChangeInProcess == "True")
+                    {
+                        for (int i = 0; i < 3; i++)
+                        {
+                            response = await _httpClient.GetAsync(uri);
+                            content = await response.Content.ReadAsStringAsync();
+                            result = JsonConvert.DeserializeObject<CyberArkCCPPassword>(content);
+                            if (result.PasswordChangeInProcess == "False")
+                            {
+                                break;
+                            }
+                            Thread.Sleep(1500);
+                        }
+                    }
+                }
+
+                if (string.IsNullOrEmpty(result.Content))
+                {
+                    var error = JsonConvert.DeserializeObject<CyberArkCCPError>(content);
+                    throw new SecureStoreException(
+                    SecureStoreException.Type.InvalidConfiguration,
+                    $"{error.ErrorCode} - {error.ErrorMsg}",
+                    normalizationException);
+                }
+
+                return result;
+            }
+            catch (Exception ex)
+            {
+                normalizationException = ex;
+            }
+
+            throw new SecureStoreException(
+                    SecureStoreException.Type.InvalidConfiguration,
+                    SecureStoresUtil.GetLocalizedResource(nameof(Resource.InvalidSecureStoreContext)),
+                    normalizationException);
+        }
+
+        private X509Certificate2Collection GetCertificateForCCPAuth(string thumbprint)
+        {
+            Exception normalizationException = null;
+
+            var certCollection = new X509Certificate2Collection();
+            try
+            {
+                using (X509Store localMachineStore = new X509Store(StoreName.My, StoreLocation.LocalMachine))
+                {
+                    localMachineStore.Open(OpenFlags.ReadOnly);
+                    //Changed `validOnly` arg to false so it can work with self-signed without CA certs.
```

Comment:
The comment is nice to have and the solution is good.
But change the comment style to be imperative _`validOnly` arg is set to false so it can work with self-signed without CA certs_